src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -1674,7 +1674,10 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             // byref assignment is also a potential write
             if (node.IsRef)
             {
-                WriteArgument(node.Right, node.Left.GetRefKind(), method: null);
+                var refKind = node.Left.Kind == BoundKind.BadExpression
```

Comment:
>node.Left.Kind == BoundKind.BadExpression [](start = 30, length = 41)

Why single out BadExpression? Consider simply always changing None returned from ```node.Left.GetRefKind()``` to ```RefKind.Ref```

Reply:
This is a special case with BadExpression because it's probably not true that it is `ref`, but since it's assumed that BadExpression has already reported a diagnostic, I've added this code to prevent cascading diagnostics. I've also added a comment to explain all this.